### PR TITLE
TT-3960 tests for OAS Endpoint page

### PR DESCRIPTION
**Pull request by maciejwojciechowski**
## Checklist

- [ ] Make sure that tests are passing in githubaction
- [ ] Check if change should be populated to other branches (master, release-3-lts, release-4.0)
- [ ] Tests are stable

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**UI tests result: success** :white_check_mark: 
Branch used: refs/pull/77/merge
Commit:  
Triggered by: pull_request (@maciejwojciechowski)
[Execution page](https://github.com/TykTechnologies/tyk-analytics-tests/actions/runs/1691615870)

File: lib/pom/Endpoints_page.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  get OAS_REMOVE_ENDPOINT_BUTTON() {return new Button_object('//ul[@class="tyk-dropdown-menu tyk-dropdown opened"]//following::span[text()="Remove Endpoint"]');}
+  get OAS_SEARCH_ENDPOINTS_INPUT() {return new Input_object('//input[@placeholder="Search Endpoints"]');}
+  get OAS_FILTER_METHOD_DROPDOWN() {return new DropDown_object('//span[contains(text(), "Show")]//following::span[1]');}
+  get OAS_REMOVE_CONFIRMATION_BUTTON() {return new Button_object('//button/span[text()="Remove Endpoint"]');}
+  get OAS_UPDATE_ENDPOINT_BUTTON() {return new Button_object('span*=Update Endpoint');}
+  
+  clickOnEndpoint(endpoint, method) {
+    let endpointUrlLink = encodeURIComponent(endpoint.split('/').join('-')) + method;
+    let endpointSelector = $(`//a[contains(@href, "${endpointUrlLink}")]`);
+    endpointSelector.click();
```
Comment (konrad-sol) [inline lib/pom/Endpoints_page.js:25]: please add any logs in here or use new Button_object(`//a[contains(@href, "${endpointUrlLink}")]`)

File: test/specs/api_designer/oas_api_endpoints_test.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+  ];
+
+  before(() => {
+    envDetails = setUpEnv();
+    login_page.open();
+    login_page.login(envDetails.userEmail, envDetails.userPassword);
+  });
+
+  it('Initially Endpoint list is empty', () => {
+    let noEndpointsMsg = $('//h2[text()="No endpoints yet."]');
```
Comment (konrad-sol) [inline test/specs/api_designer/oas_api_endpoints_test.js:47]: why not keep this selector in POM?

File: test/specs/api_designer/oas_api_endpoints_test.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    browser.refresh();
+    endpoints_page.OAS_ENDPOINTS_BUTTON.click();
+    createdEndpointsList.forEach (e => {
+      verifyDisplayedEndpoints(e, createdEndpointsList);
+    });
+  });
+
+  it('User can filter endpoints by name', () => {
+    let searchCriteria = 'ip';
+    let expectedEndpointList = [
```
Comment (konrad-sol) [inline test/specs/api_designer/oas_api_endpoints_test.js:78]: those 'let' could be 'const'

File: test/specs/api_designer/oas_api_endpoints_test.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    expect(apis_page.isApiUpdatedPopUpDisplayed()).to.be.true; 
+    browser.refresh();
+    endpoints_page.OAS_ENDPOINTS_BUTTON.click();
+    wdioExpect(removeEendpointSelector).not.toBeDisplayed();
+  });
+
+  function verifyDisplayedEndpoints(endpointObject, expectedEndpoints) {
+    let endpointUrlLink = encodeURIComponent(endpointObject.endpoint.split('/').join('-')) + endpointObject.method;
+    let endpointSelector = $(`//a[contains(@href, "${endpointUrlLink}")]`);
+    if (JSON.stringify(expectedEndpoints).includes(JSON.stringify(endpointObject))){
```
Comment (konrad-sol) [inline test/specs/api_designer/oas_api_endpoints_test.js:166]: I think that this is a place where comment (with brief explanation) will help a lot in the future:)

**Review by konrad-sol — CHANGES_REQUESTED** [review]

**Comment by Tyk-ITS** [comment]
**UI tests result: success** :white_check_mark: 
Branch used: refs/pull/77/merge
Commit: 26cbcaa4b8d7db944d0c6cfe9102d91d696ac576 
Triggered by: pull_request (@maciejwojciechowski)
[Execution page](https://github.com/TykTechnologies/tyk-analytics-tests/actions/runs/1692480544)

**Review by konrad-sol — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**UI tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 025b13deee91740cba1fdaa34d17180cc2f86d6e added tests for OAS Endpoint page
Triggered by: push (@konrad-sol)
[Execution page](https://github.com/TykTechnologies/tyk-analytics-tests/actions/runs/1692923562)

**Pull request merged**